packages/jest-config/src/normalize.js

```diff
@@ -650,8 +650,11 @@ export default function normalize(
       case 'moduleFileExtensions': {
         value = options[key];
 
-        // If it's the wrong type, it can throw at a later time
-        if (Array.isArray(value) && !value.includes('js')) {
+        if (
+          Array.isArray(value) && // If it's the wrong type, it can throw at a later time
+          options.runner === undefined && // Only require 'js' for the default jest-runner
```

Comment:
```suggestion
          (options.runner === undefined || options.runner === DEFAULT_CONFIG.runner) && // Only require 'js' for the default jest-runner
```

Reply:
to handle people doing `const {defaults} = require('jest-config');` and returning some version of that

Reply:
👍 also added it to the `throws if missing js` test